### PR TITLE
feat(pypi): changelog lookup in project_urls

### PR DESCRIPTION
**Pull request by trim21**
Closes #

### CODE REVIEW

File: lib/datasource/pypi/index.ts [hunk 1]
```diff
@@ -83,15 +83,39 @@ async function getDependency(
       }
     }
 
-    if (dep.info?.project_urls && !dependency.sourceUrl) {
+    if (dep.info?.project_urls) {
+      if (!dependency.sourceUrl) {
+        for (const [name, projectUrl] of Object.entries(
```
Comment (viceice) [inline lib/datasource/pypi/index.ts:88]: Can you unify the loops, so we don't iterate twice

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/datasource/pypi/index.ts (hunk 1 above)
Comment (Trim21) [reply lib/datasource/pypi/index.ts:88]: maybe after #6420 is merged? there would be some merge conflics

Comment (viceice) [reply lib/datasource/pypi/index.ts:88]: merged

File: lib/datasource/pypi/index.ts [hunk 2]
```diff
@@ -83,17 +83,33 @@ async function getDependency(
...
-        if (
-          lower.startsWith('repo') ||
-          lower === 'code' ||
-          lower === 'source' ||
-          github_repo_pattern.exec(projectUrl)
-        ) {
-          dependency.sourceUrl = projectUrl;
-          break;
+    if (dep.info?.project_urls) {
+      if (!dependency.sourceUrl) {
```
Comment (viceice) [inline lib/datasource/pypi/index.ts:87]: shouldn't the if combined to the inner condition?

Comment (Trim21) [reply lib/datasource/pypi/index.ts]: sorry, wrong if condition

File: lib/datasource/pypi/index.ts [hunk 3]
```diff
@@ -83,17 +84,33 @@ async function getDependency(
...
+        if (
+          [
+            'changelog',
+            'change log',
+            'changes',
+            'release notes',
+            'news',
+            "what's new",
+          ].includes(lower) ||
+          changelogFilenameRegex.exec(lower)
```
Comment (viceice) [inline lib/datasource/pypi/index.ts:110]:
```suggestion
          !dependency.changelogUrl &&
          ([
            'changelog',
            'change log',
            'changes',
            'release notes',
            'news',
            "what's new",
          ].includes(lower) ||
          changelogFilenameRegex.exec(lower))
```

Only assign if undefined

**Review by viceice — APPROVED** [review]

**Pull request merged**